### PR TITLE
Revised Concept Map

### PR DESCRIPTION
**Pull request by neenjaw**
Ongoing design changes prompting an update  in multiple areas.

1: progress bar is now replaced with a "dot state" to represent the exercise status of the exercises associated with the concept
2: reintroduction of locked/unlocked state

## WIP rough styling

Screenshot from 2021-03-21 15-59-50

### CODE REVIEW

**Comment by neenjaw** [comment]
@iHiD this doesn't have any backend changes, but I think it contains all of the potential state combinations if you wanted to push commits to style it on this branch

**Comment by neenjaw** [comment]
Things left to do (a wip):
- update tests
- ____

File: app/commands/user_track/generate_exercise_status_mapping.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        pluck("track_concepts.slug", "exercises.slug").
+        each do |concept_slug, exercise_slug|
+        next if concept_slug.nil?
+        next if exercise_slug.nil?
+
+        mapping[concept_slug] << exercise_slug
+      end
+
+      # TOOD: Change to practice link and remove limit
+      Exercise::Prerequisite.joins(:exercise, :concept).
```
Comment (iHiD) [inline app/commands/user_track/generate_exercise_status_mapping.rb]: @neenjaw Once you rebase, you can change this to `Exercise::PracticedConcept` and remove the `limit` two lines below.

Comment (neenjaw) [reply app/commands/user_track/generate_exercise_status_mapping.rb]: done

**Comment by neenjaw** [comment]
Test should be passing now

**Pull request merged**